File: JavaScriptSolutions/007_nthPrime.js
const isPrime = num => {
    if (num <= 1) return false;
    if (num === 2 || num === 3) return true;
    if (num % 2 === 0) return false;
        
    for (let i = 3; i <= Math.sqrt(num); i += 2) {
        if(num % i === 0) return false;
    }

    return true;
}

const nthPrime = rank => {
    let test = 0;
    for (let num = 0; test <= rank; num++) {
        if (isPrime(num)) {
            test++;
            if (test === rank) {
                return num;     
            }
        }
    }
}

File: PythonSolutions/007_nthPrime.py
def isPrime(n):
    if n < 2:
        return False
    for i in range(2, int(n**0.5)+1):
        if n % i == 0:
            return False
    return True

def nthPrime(n):
    count = 0
    num = 1
    while count < n:
        num += 1
        if isPrime(num):
            count += 1
    return num

if __name__ == '__main__':
    n = int(input("Enter the value of n: "))
    print(nthPrime(n))